### ISSUE TITLE
Bug: `windows-bindgen` unconditionally emits unneccessary items

### ISSUE THREAD

**Issue by Jake-Shadle** [microsoft/windows-rs#2408]
### Which crate is this about?

other (please share in the comments)

### Crate version

`windows-bindgen 0.47.0`

### Summary

Using `windows_bindgen::standalone` unconditionally emits items that might not be required by the items requested by the user.

### Toolchain version/configuration

```
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/jake/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu (default)
nightly-2022-08-29-x86_64-unknown-linux-gnu
nightly-2022-10-01-x86_64-unknown-linux-gnu
nightly-2022-10-29-x86_64-unknown-linux-gnu
nightly-2022-12-18-x86_64-unknown-linux-gnu
nightly-2023-01-30-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
1.65.0-x86_64-unknown-linux-gnu
1.66.0-x86_64-unknown-linux-gnu
1.66.1-x86_64-unknown-linux-gnu
1.67.0-x86_64-unknown-linux-gnu
1.68.0-x86_64-unknown-linux-gnu

installed targets for active toolchain
--------------------------------------

aarch64-apple-darwin
wasm32-unknown-unknown
x86_64-pc-windows-msvc
x86_64-unknown-linux-gnu
x86_64-unknown-linux-musl

active toolchain
----------------

stable-x86_64-unknown-linux-gnu (default)
rustc 1.68.2 (9eb3afe9e 2023-03-27)
```

### Reproducible example

```rust
fn generate_bindings() {
    let apis = [
        "Windows.Win32.System.Console.SetConsoleTextAttribute",
    ];

    let bindings = windows_bindgen::standalone(&apis);
    std::fs::write("src/windows_term/bindings.rs", bindings).expect("failed to generate bindings");
}
```

### Crate manifest

```toml
[package]
name = "console"

[dev-dependencies]
windows-bindgen = "0.47"
```

### Expected behavior

Only the items requested are emitted.

### Actual behavior

```rust
pub type HRESULT = i32;
pub type HSTRING = *mut ::core::ffi::c_void;
pub type IUnknown = *mut ::core::ffi::c_void;
pub type IInspectable = *mut ::core::ffi::c_void;
pub type PSTR = *mut u8;
pub type PWSTR = *mut u16;
pub type PCSTR = *const u8;
pub type PCWSTR = *const u16;
pub type BSTR = *const u16;
#[repr(C)]
pub struct GUID {
    pub data1: u32,
    pub data2: u16,
    pub data3: u16,
    pub data4: [u8; 8],
}
impl GUID {
    pub const fn from_u128(uuid: u128) -> Self {
        Self {
            data1: (uuid >> 96) as u32,
            data2: (uuid >> 80 & 0xffff) as u16,
            data3: (uuid >> 64 & 0xffff) as u16,
            data4: (uuid as u64).to_be_bytes(),
        }
    }
}
impl ::core::marker::Copy for GUID {}
impl ::core::clone::Clone for GUID {
    fn clone(&self) -> Self {
        *self
    }
}
```

Is emitted in addition to whatever bindings were actually requested.

### Additional comments

[This is the emitted block](https://github.com/microsoft/windows-rs/blob/ec95c1e6ac1eeabb6f8c91c41d0197ecf43fa75c/crates/libs/bindgen/src/lib.rs#L214-L242). Presumably fixing #2407 would mean this could be completely removed as it would not be necessary to emit those items unless they were actually referenced by 1 or more of the items requested by the user.

**Comment by kennykerr** [comment]
These are considered the "core" types by the `bindgen` crate and what it expects of `windows` and `windows-sys` respectively, but yes at least for the sys-style bindings we should be able to infer these along with #2407.

**Comment by kennykerr** [comment]
Here's the fix for this: #2421
